src/main/scala/edu/ie3/mobsim/model/ChargingBehavior.scala

```diff
@@ -74,39 +75,51 @@ object ChargingBehavior extends LazyLogging {
         logger.debug(
           s"${ev.getId} arrives at charging hub and wants to start charging."
         )
-        ev.getDestinationPoi.nearestChargingStations.keys.headOption
-          .map(_.getUuid)
+        (
+          ev.destinationPoi.nearestChargingStations.keys.headOption
+            .map(_.uuid),
+          None
+        )
       } else {
         /* Update charging prices memory of EV to have a reference for the prices of specific charging stations */
-        ev.getDestinationPoi.nearestChargingStations.keys.foreach { cs =>
-          currentPricesAtChargingStations
-            .get(cs.getUuid)
-            .map(ev.updateChargingPricesMemory)
-        }
+
+        val prices = immutable.Queue.from(
+          ev.destinationPoi.nearestChargingStations.keys.map { cs =>
+            currentPricesAtChargingStations(cs.uuid)
+          }
+        )
```

Comment:
Could be simplified with the `to` method:

```suggestion
        val prices = ev.destinationPoi.nearestChargingStations.keys
          .map { cs =>
            currentPricesAtChargingStations(cs.uuid)
          }
          .to(immutable.Queue)
```